lightning/src/ln/interactivetxs.rs

```diff
@@ -876,13 +1036,19 @@ impl InteractiveTxConstructor {
 		entropy_source: &ES, channel_id: ChannelId, feerate_sat_per_kw: u32, is_initiator: bool,
 		funding_tx_locktime: AbsoluteLockTime,
 		inputs_to_contribute: Vec<(TxIn, TransactionU16LenLimited)>,
-		outputs_to_contribute: Vec<TxOut>,
+		outputs_to_contribute: Vec<TxOut>, intended_new_funding_output_script: ScriptBuf,
+		intended_local_contribution_satoshis: u64,
```

Comment:
Not really, because of fees. The share of the output cannot be higher than the share of the inputs, but it can be lower, the rest being used for fees (fees have to meet a minimum).